### PR TITLE
test: Add public API contract test, improve docs site type formatting

### PR DESCRIPTION
**Pull request by mattsjones**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,11 +9,12 @@
   "scripts": {
     "check-integrity": "yarn check --integrity && echo 'Ignore paths from lint-staged'",
     "pretest": "sku configure",
-    "test": "yarn check --integrity && yarn lint && yarn build",
+    "test": "yarn check --integrity && yarn lint && yarn build && yarn contract-test",
+    "contract-test": "sku test",
```
Comment (michaeltaranto) [inline package.json]: Given the name of this command i would recommend explicitly passing the path to the contract test here.

File: lib/componentDocs/generateComponentDocs.js [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
-});
-
-const omitDocWithNoProps = ({ props = {} }) => Object.keys(props).length > 0;
-const omitNonComponentDocs = ({ displayName }) => /^[A-Z]/.test(displayName);
-
-(async () => {
-  const componentDocs = await globsToReactDocgen({
-    include: ['lib/themes/**/*.{ts,tsx}', 'lib/components/**/*.{ts,tsx}'],
-    exclude: [
-      'lib/components/private/**/*',
```
Comment (michaeltaranto) [inline lib/componentDocs/generateComponentDocs.js]: Love that we dont need to manage exclusions now, its solely testing the public api 🙌

File: generate-component-docs/__snapshots__/generate.test.ts.snap [hunk 3]
```diff
@@ -0,0 +1,6618 @@
...
+    },
+  },
+}
+`;
+
+exports[`Public API Box 1`] = `
+Object {
+  "props": Object {
+    "about": Object {
+      "propName": "about",
```
Comment (michaeltaranto) [inline generate-component-docs/__snapshots__/generate.test.ts.snap]: Any thoughts on being able to apply the breaks on components like this and not traverse every html prop?

Comment (mattsjones) [reply generate-component-docs/__snapshots__/generate.test.ts.snap]: I thought about it. It's probably doable. But it's still part of our contract. Was thinking we could see how it goes.

File: generate-component-docs/__snapshots__/generate.test.ts.snap [hunk 4]
```diff
@@ -0,0 +1,6618 @@
...
+      "required": false,
+      "type": "string",
+    },
+    "rowSpan": Object {
+      "propName": "rowSpan",
+      "required": false,
+      "type": "number",
+    },
+    "rows": Object {
+      "propName
```
Comment (michaeltaranto) [inline generate-component-docs/__snapshots__/generate.test.ts.snap]: `ReactNode` might be something else to not unpack, from a docs site perspective things can get noisy here.

**Pull request merged**